sdk/src/commitment_config.rs

```diff
@@ -0,0 +1,36 @@
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct CommitmentConfig {
+    pub commitment: CommitmentLevel,
+}
+
+impl Default for CommitmentConfig {
+    fn default() -> Self {
+        CommitmentConfig {
+            commitment: CommitmentLevel::Max,
+        }
+    }
+}
+
+impl CommitmentConfig {
+    pub fn recent() -> Self {
+        Self {
+            commitment: CommitmentLevel::Recent,
+        }
+    }
+
+    pub fn ok(&self) -> Option<Self> {
+        if self == &Self::default() {
+            None
+        } else {
+            Some(self.clone())
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub enum CommitmentLevel {
```

Comment:
Do we need `CommitmentConfig` at all?   `CommitmentLevel` is the real meat and `CommitmentConfig` just wraps it.   Did you do this in the expectation of adding other fields to `CommitmentConfig` later?

Reply:
Yeah it makes it slightly easier to extend or make more configurable later, but mostly I thought the rpc arg was better serialized as {"commitment":"max"} than just "max" (which is how the enum by itself would appear)

Reply:
wfm, :shipit: